### PR TITLE
SDKS-2782 cookiename/AM URL/Realm are mandatory field

### PR DESCRIPTION
**Pull request by jeyanthanperiyasamy**
# JIRA Ticket

Please link jira ticket here

# Description

1.The idea is to keep by default the cookiename and realm has value if the user didnt add this
 cookiename = iplanetdirectorypro 
 realm = root

2. validate the AMURL, cookiename, realm is not blank  or empty

### CODE REVIEW

File: forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt [hunk 1]
```diff
@@ -66,20 +66,20 @@ internal class ConfigHelper {
 
             return FROptionsBuilder.build {
                 server {
-                    url = sharedPreferences.getString(ConfigHelper.url, null) ?: ""
-                    realm = sharedPreferences.getString(ConfigHelper.realm, null) ?: ""
-                    cookieName = sharedPreferences.getString(ConfigHelper.cookieName, null) ?: ""
+                    url = sharedPreferences.getString(ConfigHelper.url, null) ?: context.getString(R.string.forgerock_url)
```
Comment (witrisna) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt:69]: Can we use the defValue sharedPreferences.getString(value, defValue) for the null case.
sharePreferences.getString(ConfigHelper.url, context.getString(R.string.forgerock_url)

**Review by witrisna — APPROVED** [review]

File: forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt (hunk 1 above)
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt:69]: Update copyright

File: forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt [hunk 2]
```diff
@@ -34,14 +34,21 @@ data class FROptions(val server: Server,
                     && old?.logger == new?.logger
         }
     }
+    @Throws(IllegalArgumentException::class)
```
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/FROptions.kt:37]: Update copyright

File: forgerock-auth/src/test/java/org/forgerock/android/auth/ConfigHelperTest.kt [hunk 3]
```diff
@@ -68,69 +68,141 @@ class ConfigHelperTest {
             urlPath {
                 revokeEndpoint = "https://revoke"
                 endSessionEndpoint = "https://endsession"
+                sessionEndpoint = "https://sessionEndpoint"
```
Comment (rodrigoareis) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/ConfigHelperTest.kt:71]: Update copyright

File: forgerock-auth/src/test/java/org/forgerock/android/auth/FROptionTest.kt [hunk 4]
```diff
@@ -59,10 +59,39 @@ class FROptionTest {
         assertTrue(option.logger.customLogger == logger)
     }
 
+    @Test(expected = IllegalArgumentException::class)
```
Comment (rodrigoareis) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/FROptionTest.kt:62]: Update copyright

File: forgerock-auth/src/main/java/org/forgerock/android/auth/FRAuth.java [hunk 5]
```diff
@@ -47,6 +47,8 @@ public static synchronized void start(Context context, @Nullable FROptions optio
         if(!started || !FROptions.equals(cachedOptions, options)) {
             started = true;
             FROptions currentOptions = ConfigHelper.load(context, options);
+            //Validate (AM URL, Realm, CookieName) is not Empty. If its empty will throw IllegalArgumentException.
+            currentOptions.validateConfig();
```
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/FRAuth.java:51]: Are we throwing the new exception here?

**Review by spetrov — DISMISSED** [review]
Please, put a note in CHANGELOG.md about this change. 
Other than that the changes look good to me!

File: forgerock-auth/src/main/java/org/forgerock/android/auth/FRAuth.java (hunk 5 above)
Comment (jeyanthanperiyasamy) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/FRAuth.java:51]: yes , we are throwing a new exception, its intentional when the user supplies empty url, realm, cookiename

File: forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt (hunk 1 above)
Comment (jeyanthanperiyasamy) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/ConfigHelper.kt:69]: no change required. already discussed

**Review by rodrigoareis — DISMISSED** [review]
Changes looks good to me

**Review by spetrov — APPROVED** [review]
LGTM! 👍🏻

**Review by spetrov — APPROVED** [review]

**Pull request merged**